server/registry_content_type_test.go

```diff
@@ -0,0 +1,320 @@
+package server
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"github.com/armory-io/go-commons/awaitility"
+	armoryhttp "github.com/armory-io/go-commons/http"
+	"github.com/armory-io/go-commons/logging"
+	"github.com/armory-io/go-commons/management/info"
+	"github.com/armory-io/go-commons/metadata"
+	metrics2 "github.com/armory-io/go-commons/metrics"
+	"github.com/armory-io/go-commons/server/serr"
+	"github.com/go-playground/validator/v10"
+	"github.com/golang/mock/gomock"
+	"github.com/samber/lo"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/suite"
+	"github.com/uber-go/tally/v4"
+	"go.uber.org/fx/fxtest"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+	"gopkg.in/yaml.v2"
+	"io"
+	"net"
+	"net/http"
+	"testing"
+	"time"
+)
+
+type (
+	ContentTypesTestSuite struct {
+		log *zap.SugaredLogger
+		suite.Suite
+		controller Controller
+		lc         *fxtest.Lifecycle
+		client     *http.Client
+		baseUrl    string
+	}
+
+	contentTypeController struct {
+	}
+
+	testTimer struct {
+	}
+)
+
+func (d *contentTypeController) Handlers() []Handler {
+	return []Handler{
+		NewHandler(d.readyHandler, HandlerConfig{
```

Comment:
Thank you for adding these much needed test cases!
Questions:
1. Should we add a Handler where it is not specified what it consumes and produces? 
2. If it is unspecified, should that handler consume and produce `application/json` by default? 
3. Would it be in direct conflict with the `dedicatedJsonEndpoint` handler? 
4. Is consumes `*/*` the same thing as omitting consumes altogether?

Reply:
if we don't specify any consumes or produces - it would default to application/json 
in such case, if i get rid of the */* it would conflict the one i've created with application/json explicitly - the server won't start